src/interview/interview.service.ts

```diff
@@ -0,0 +1,145 @@
+  private readonly dataSource: DataSource;
+  @Inject(forwardRef(() => FiltrationService))
+  private readonly filtrationService: FiltrationService;
+
+  async createInterview(
+    studentId: string,
+    user: User,
+  ): Promise<InterviewResponse> {
+    const newInterview = new Interview();
+    const hr = (await this.usersService.findOne(user.id)).hr;
+    const student = await this.studentsService.findOne(studentId);
+
+    if ((await this.getCountInterview(hr)) >= hr.maxReservedStudents) {
+      throw new HttpException(
+        'Osiągnięto maksymalną liczbę rezerwacji na rozmowy kwalifikacyjne',
+        HttpStatus.CONFLICT,
+      );
+    }
+
+    if (student.status === StudentStatus.HIRED) {
+      throw new HttpException(
+        'Student został już zatrudniony. Nie można dodać go do rozmowy kwalifikacyjnej',
+        HttpStatus.CONFLICT,
+      );
+    }
+    if (student.status === StudentStatus.CONVERSATION) {
+      throw new HttpException(
+        'Student jest w trakcie rozmowy. Nie można dodać go do rozmowy kwalifikacyjnej',
+        HttpStatus.CONFLICT,
+      );
+    }
+
+    newInterview.hr = hr;
+    newInterview.student = student;
+
+    if (student.status === StudentStatus.AVAILABE) {
+      student.status = StudentStatus.CONVERSATION;
+      await student.save();
+    }
+
+    return await newInterview.save();
+  }
+
+  async getCountInterview(hr): Promise<number> {
+    const interviews = await this.findInterview(hr);
+    return interviews.length;
+  }
+
+  async findInterview(hr): Promise<InterviewResponse[]> {
+    return await Interview.find({ where: { hr } });
+  }
+
+  async findAllInterview(searchOptions: SearchOptionsDto, user) {
+    const queryBuilder = await this.dataSource
+      .getRepository(Interview)
+      .createQueryBuilder('interview')
+      .leftJoinAndSelect('interview.student', 'student')
+      .leftJoinAndSelect('interview.hr', 'hr')
+      .skip(searchOptions.skip)
+      .take(searchOptions.take);
+
+    switch (user.role) {
+      case UserRole.HR:
+        const hr = (await this.usersService.findOne(user.id)).hr;
+        queryBuilder.andWhere('interview.hr = :hrId', { hrId: hr.id });
+        break;
+      case UserRole.STUDENT:
+        const student = (await this.usersService.findOne(user.id)).student;
+        queryBuilder.andWhere('interview.student = :studentId', {
+          studentId: student.id,
+        });
+    }
+
+    const filterQueryBuilder =
+      await this.filtrationService.filterStudentPreferences(
+        searchOptions,
+        queryBuilder,
+      );
+
+    const itemCount = await filterQueryBuilder.getCount();
+    const { entities } = await filterQueryBuilder.getRawAndEntities();
+    const pageMetaDto = new PageMetaDto({ searchOptions, itemCount });
+    return { data: entities, meta: { ...pageMetaDto } };
+  }
+
+  async removeInterview(
+    studentId: string,
+    user: any,
+  ): Promise<InterviewResponse[]> {
+    let hr;
+    user.role === UserRole.HR
```

Comment:
Oj proszę, nie tak. Ultra nieczytelne. Tak się nie używa operatora trójargumentowego.
`hr = user.role === ... ? await this... : user`